### DISCUSSION TITLE
problem running jsPsych on Pavlovia

### DISCUSSION THREAD

**Discussion by IrisxBi** [jspsych/jsPsych#1090] [Q&A] [answered]
Hi! 
I'm new to jsPsych and online experiments so I'd be grateful for any pointers relating to this issue: 

I have a jsPsych experiment that runs properly from my computer. I am trying to run it on Pavlovia, so as per the instructions on pavlovia.com (https://pavlovia.org/docs/experiments/create-jsPsych). I loaded in the jspsych-pavlovia plugin and jquery at the head of my file: 

    <script type="text/javascript" src="jquery-3.5.1.js"></script>
    <script type="text/javascript" src="jspsych-pavlovia-3.0.0.js"></script>

And also added these lines to the beginning and end of my timeline: 

//added to beginning of timeline
/* init connection with pavlovia.org */
var pavlovia_init = {
  type: "pavlovia",
  command: "init"
};
timeline.push(pavlovia_init);

//added to end of timeline
var pavlovia_finish = {
  type: "pavlovia",
  command: "finish"
  };
timeline.push(pavlovia_finish);

However, after adding these, I can no longer run it from my computer or from pavlovia.com, and get the following error: 
 Access to XMLHttpRequest at 'file:///experiment/config.json?json' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https

I'm not sure that this is a jsPsych-specific issue, but I tried following some of the solutions online addressing a similar issue for experiments that aren't coded in jsPsych and haven't had much luck. Does anyone happen to have run into a similar issue, or has any pointers?

**Comment by becky-gilbert** [comment, marked as answer]
Hi @IrisxBi, I'm not a Pavlovia user so I'm not sure I can help, and I'm hoping someone else can jump in! I just wanted to say that the CORS error might be expected when running the file locally, after it's been set up to run on Pavlovia. Do you get a different error when you try to run your experiment on the Pavlovia site? And have you tried contacting Pavlovia support?

**Reply by IrisxBi** [reply]
Hi @becky-gilbert , thanks for answering!
 Turns out it was a minor issue - when running on the Pavlovia site, the filepaths that load in the jsPsych files needed to match the name of the folder exactly, ie. there had to be a capital letter in my filepath: 
<script src=jsPsych/jspsych.js></script> 
because the folder was named "jsPsych". This didn't seem to be a problem when running locally.

**Reply by becky-gilbert** [reply]
Aha! Thanks for the update and glad you figured it out! 

Your answer might be useful for non-Pavlovia users too, because may web servers run on Unix/Linux, which use case-sensitive file paths. Whereas Windows file paths are case insensitive, which means file paths with incorrect capitalization will still work if you're using Windows locally. Maybe we should add this to a troubleshooting section of the jsPsych documentation, because it might be a fairly common problem.
